tests/garage/tf/policies/test_categorical_mlp_policy_with_model.py

```diff
@@ -0,0 +1,49 @@
+"""This script creates a unittest that tests qf-derived policy."""
+import pickle
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.envs import TfEnv
+from garage.tf.policies import CategoricalMLPPolicyWithModel
+from tests.fixtures import TfGraphTestCase
+from tests.fixtures.envs.dummy import DummyDiscreteEnv
+
+
+class TestCategoricalMLPPolicyWithModel(TfGraphTestCase):
+    def setUp(self):
+        super().setUp()
+        self.env = TfEnv(DummyDiscreteEnv())
+        self.policy = CategoricalMLPPolicyWithModel(env_spec=self.env.spec)
+        self.env.reset()
+
+    def test_categorical_mlp_policy_with_model_get_action(self):
+        obs, _, _, _ = self.env.step(1)
+        action, _ = self.policy.get_action(obs)
+        assert self.env.action_space.contains(action)
+        actions, _ = self.policy.get_actions([obs])
+        for action in actions:
+            assert self.env.action_space.contains(action)
+
+    def test_categorical_mlp_policy_with_model_is_pickleable(self):
+        with tf.Session(graph=tf.Graph()):
+            p = pickle.dumps(self.policy)
+            policy_pickled = pickle.loads(p)
+            obs, _, _, _ = self.env.step(1)
+            action, _ = policy_pickled.get_action(obs)
+            assert self.env.action_space.contains(action)
```

Comment:
you need to assert something stronger here -- that the pre-pickling and post-pickling policy produce identical output for a given input